### PR TITLE
Create a version of SAM(SON) code from this repo that successfully compiles

### PR DESCRIPTION
**Pull request by jatkinson1000**
This addresses the fact that the code provided was unable to be build or run, and was provided without resources for making, running or a readme.

- [x] Restructure to the format required for automated building
- [x] Provide functional Makefile and Build script
- [x] Provide documentation on how to build
- [x] Provide inputs for running

### CODE REVIEW

**Comment by jatkinson1000** [comment]
I have managed to successfully build using (close to) Janni's Makefile and Build script on Cheyenne, and successfully run the executable fresh (had to change to no-restart and remove some binaries).

I added a simple readme to describe how to build/run.

This is now ready to merge after review, to be followed by a native run and a run with our new code.

File: SAMSON_NN/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+It may also contain:
+
+- `grd`
+- `lsf`
+- `sfc`
+- `rad`
+
+as described in the SAM documentation.
+
+The executable can be run using `mpirun`. E,g, to run on 8 processors:
```
Comment (dorchard) [inline SAMSON_NN/README.md:46]: Surely this means that mpi is not optional (as described further up the in README)- or can we do a non-MPI run?

Comment (jatkinson1000) [reply SAMSON_NN/README.md:46]: hhmmmmmm SAM Docs say you can build without (switch `task_util_MPI.f90` for `task_util_NOMPI.f90`) but then only provide this as a way to run...

Maybe I'll just say it's required for our purposes.

Comment (jatkinson1000) [reply SAMSON_NN/README.md:46]: Updated accordingly.
Interestingly I tested using the `NOMPI` build initially, so I assume using `mpirun` just launches 8 threads of the single thread code?
Have tested MPI build as well now.

**Pull request merged**